src/main/java/Task.java

```diff
@@ -0,0 +1,70 @@
+package main.java;
+
+import java.util.ArrayList;
+
+/**
+ * A Task stores the name of the task, whether it is completed,
+ * and subtasks (optional)
+ */
+public class Task {
+
+    private String taskName;
+    private boolean completed;
+    private ArrayList<Task> subTasks;
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.completed = false;
+        this.subTasks = new ArrayList<Task>();
+    }
+
+    public void renameTask(String newName) {
+        this.taskName = newName;
+    }
+
+    public void complete() {
+        this.completed = true;
+    }
+
+    public void undoComplete() {
+        this.completed = false;
+    }
```

Comment:
I would suggest merging those methods with one setter for `completed`

Reply:
Ok. And I will also rename other setters.